dandiapi/api/tests/factories.py

```diff
@@ -116,7 +151,30 @@ class AssetMetadataFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = AssetMetadata
 
-    metadata = factory.Faker('pydict', value_types=['str', 'float', 'int'])
+    @factory.lazy_attribute
+    def metadata(self):
+        return {
+            'schemaVersion': '0.3.1',
+            'encodingFormat': 'application/x-nwb',
+            'publishedBy': {
+                'id': 'urn:uuid:08fffc59-9f1b-44d6-8e02-6729d266d1b6',
+                'name': 'DANDI publish',
+                'startDate': '2021-05-18T19:58:39.310338-04:00',
+                'endDate': '2021-05-18T19:58:39.310361-04:00',
```

Comment:
these two should reflect the start and end of the publishing activity.

Reply:
So the whole `publishedBy` block needs to be populated at validation time without modifying the draft versions, then included in the newly published metadata? Are there any other fields that need this treatment?

Reply:
i believe these are all the fields that change on the publish trigger:
`id, publishedBy, datePublished, url, version, doi, manifestLocation, citation`

i think almost all of them can be set on version save as a function of the clicking/posting publish. the publishedBy has an endDate, which should correspond to datePublished

for all datetimes i was using `datetime.astimezone(datetime.now()).isoformat()` to preserve timezone info.

Reply:
I will address these in a new PR, I don't want to dilute this one too much.